packages/backend-app-api/src/wiring/ServiceRegistry.ts

```diff
@@ -146,6 +146,43 @@ export class ServiceRegistry implements EnumerableServiceHolder {
     }
   }
 
+  #checkForCircularDeps(factory: InternalServiceFactory, pluginId: string) {
+    const head = factory;
+
+    const nodes = Object.values(factory.deps);
+
+    const depChain: Array<ServiceRef<unknown, 'plugin' | 'root'>> = [
+      head.service,
+    ];
+
+    let node = nodes.shift();
+
+    if (node) {
+      depChain.push(node);
+    }
+
+    while (!!node && node.id !== head.service.id) {
+      const nodeFactory = this.#providedFactories.get(node.id);
+      const nodeDeps = nodeFactory?.deps;
+      if (nodeDeps) {
+        nodes.unshift(...Object.values(nodeDeps));
+      }
+      node = nodes.shift();
+      if (node) {
+        depChain.push(node);
```

Comment:
Always pushing a node here seems like it would result in incorrect chains in many cases. For example `A -> B,C` and `C -> A`. I think that'd tell you that the chain is `A -> B -> C -> A` right?

Reply:
Added a unit test to cover this scenario